ext/pdo_odbc/odbc_driver.c

```diff
@@ -344,6 +344,30 @@ static bool odbc_handle_set_attr(pdo_dbh_t *dbh, zend_long attr, zval *val)
 			}
 			H->assume_utf8 = bval;
 			return true;
+		case PDO_ATTR_AUTOCOMMIT:
+			if (!pdo_get_bool_param(&bval, val)) {
+				return false;
+			}
+			if (dbh->in_txn) {
+				pdo_raise_impl_error(dbh, NULL, "HY000", "Cannot change autocommit mode while a transaction is already open");
+				return false;
+			}
+			if (dbh->auto_commit ^ bval) {
+				dbh->auto_commit = bval;
+				RETCODE rc = SQLSetConnectAttr(
+					H->dbc,
+					SQL_ATTR_AUTOCOMMIT,
+					dbh->auto_commit ? (SQLPOINTER) SQL_AUTOCOMMIT_ON : (SQLPOINTER) SQL_AUTOCOMMIT_OFF,
+					SQL_IS_INTEGER
+				);
+				if (rc != SQL_SUCCESS) {
```

Comment:
According to the documentation this could also return `SQL_SUCCESS_WITH_INFO`.

Reply:
According to this document, SQL_SUCCESS_WITH_INFO is returned when there is a warning.
https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/sqlsetconnectattr-function?view=sql-server-ver16#diagnostics

Also, this API is used for begin, commit, etc., but all of them expect SQL_SUCCESS, and no bugs have been reported so far, so it seems safe to assume that SQL_SUCCESS will be returned in case of success. is.
https://github.com/php/php-src/blob/1e7c64bd4dc65e3361e3b86d90aac1b116d25f69/ext/pdo_odbc/odbc_driver.c#L267